apps/web/src/pages/integrations/components/UpdateIntegrationSidebarHeader.tsx

```diff
@@ -67,7 +100,19 @@ export const UpdateIntegrationSidebarHeader = ({
                 </div>
               }
               middlewares={{ flip: false, shift: false }}
+              position="bottom-end"
             >
+              {canMarkAsPrimary && (
+                <Dropdown.Item
+                  onClick={() => {
+                    makePrimaryIntegration({ id: provider.integrationId });
+                  }}
+                  icon={<StarEmpty />}
+                  disabled={isLoading || isMarkingPrimary}
+                >
+                  Mark as primary
+                </Dropdown.Item>
+              )}
```

Comment:
only email/sms integrations could be marked as primary and also the integration that is not already primary ;)